src/Approximations/decompositions.jl

```diff
@@ -59,19 +62,159 @@ projections.
+For example:
+
+```jldoctest decompose_examples
+julia> import LazySets.Approximations:decompose
+julia> S = Ball2(zeros(4), 1.);
+julia> array(decompose(S))
+2-element Array{LazySets.Hyperrectangle{Float64},1}:
+ LazySets.Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
+ LazySets.Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
+```
+
+Other block sizes can be specified using the `blocks` option, that refers to
+each block size of the partition:
+
+```jldoctest decompose_examples
+julia> array(decompose(S, blocks=[1, 3]))
+2-element Array{LazySets.Hyperrectangle{Float64},1}:
+ LazySets.Hyperrectangle{Float64}([0.0], [1.0])
+ LazySets.Hyperrectangle{Float64}([0.0, 0.0, 0.0], [1.0, 1.0, 1.0])
+
+julia> array(decompose(S, blocks=[4]))
+1-element Array{LazySets.Hyperrectangle{Float64},1}:
+ LazySets.Hyperrectangle{Float64}([0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0])
+```
+
+#### Different set types
+
+We can also decompose using polygons in constraint representation, through the
+`set_type` optional argument:
+
+```jldoctest decompose_examples
+julia> [ai isa HPolygon for ai in array(decompose(S, set_type=HPolygon))]
+2-element Array{Bool,1}:
+ true
+ true
+
+julia> [ai isa Interval for ai in array(decompose(S, set_type=Interval))]
+4-element Array{Bool,1}:
+ true
+ true
+ true
+ true
+```
+
+However, if you need to specify different set types for different blocks, the
+interface presented so far does not apply. In the paragraph
+*Advanced different set types input* we explain `block_types`, useful precisely
+for that purpose.
+
+#### Refining the decomposition
+
+The ``ɛ`` option can be used to refine, that is obtain a more accurate decomposition
+in those blocks where `HPolygon` types are used, and it relies on the iterative
+refinement algorithm provided in the `Approximations` module.
+
+To illustrate this, consider the unit 4D ball in the 2-norm. Using smaller ``ɛ``
+implies a better precision, thus more constraints in each 2D decomposition:
+
+```jldoctest decompose_examples
+julia> S = Ball2(zeros(4), 1.);
+julia> d(ε, bi) = array(decompose(S, set_type=HPolygon, ε=ε))[bi]
+julia> [length(constraints_list(d(ε, 1))) for ε in [Inf, 0.1, 0.01]]
+
+3-element Array{Int64,1}:
+  4
+  8
+ 32
+```
+
+#### Advanced different set types input
+
+We can define different set types for different blocks, using the
+optional `block_types` input argument. It is a dictionary where the keys correspond
+to set types, and the values correspond to the blocks, namely the initial and final
+block variables should be given.
+
+For example:
+
+```jldoctest decompose_examples
+julia> S = Ball2(zeros(3), 1.);
+julia> array(decompose(S, block_types=Dict(Interval=>[1:1], Hyperrectangle=>[2:3])))
+
+2-element Array{LazySets.Hyperrectangle{Float64},1}:
+ LazySets.Interval{Float64,IntervalArithmetic.Interval{Float64}}([-1, 1])
+ LazySets.Hyperrectangle{Float64}([0.0, 0.0], [1.0, 1.0])
+```
+
+We can the options, and the ``ε`` is passed to the `HPolygon` if it corresponds:
```

Comment:
We can additionally pass ``ε``, which is automatically used for each `HPolygon` type block.